Code/GraphMol/ChemReactions/ReactionUtils.cpp

```diff
@@ -172,19 +172,14 @@ bool isReactionTemplateMoleculeAgent(const ROMol &mol, double agentThreshold) {
 
 namespace {
 
-void getMappingNumAtomIdxMapReactants(
-    ChemicalReaction *rxn, std::map<int, Atom::ChiralType> &reactantMapping) {
-  for (MOL_SPTR_VECT::const_iterator reactIt = rxn->beginReactantTemplates();
-       reactIt != rxn->endReactantTemplates(); ++reactIt) {
-    for (ROMol::AtomIterator reactAtomIt = (*reactIt)->beginAtoms();
-         reactAtomIt != (*reactIt)->endAtoms(); ++reactAtomIt) {
-      if (!(*reactAtomIt)->hasProp("molAtomMapNumber")) {
-        continue;
-      }
+void getMappingNumAtomIdxMapReactants(ChemicalReaction *rxn, std::map<int,Atom::ChiralType>& reactantMapping){
+  for(MOL_SPTR_VECT::const_iterator reactIt=rxn->beginReactantTemplates();
```

Comment:
RDKit is almost older than smart pointers (and certainly older than ones that are thread safe).  That being said, newer code should probably use references.

PRECONDITIONING on null pointers (for pointer interfaces) protects against seg-faults and also is good debugging practice.  We don't want the rdkit based web server to go down because of a null dereference.